tests/src/tracing/eventpipe/providervalidation/providervalidation.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.Tools.RuntimeClient;
+using Microsoft.Diagnostics.Tracing;
+using Tracing.Tests.Common;
+
+namespace Tracing.Tests.ProviderValidation
+{
+    public sealed class MyEventSource : EventSource
+    {
+        private MyEventSource() {}
+        public static MyEventSource Log = new MyEventSource();
+        public void MyEvent() { WriteEvent(1, "MyEvent"); }
+    }
+
+    public class ProviderValidation
+    {
+        private static Dictionary<string, int> _expectedEventCounts = new Dictionary<string, int>()
+        {
+            { "MyEventSource", 1000 },
+            { "Microsoft-Windows-DotNETRuntimeRundown", -1 },
+            { "Microsoft-DotNETCore-SampleProfiler", -1 }
+        };
+
+        private static Action _eventGeneratingAction = () => 
+        {
+            Thread.Sleep(500);
+            foreach (var _ in Enumerable.Range(0,1000))
+            {
+                MyEventSource.Log.MyEvent();
+            }
+        };
+
+        public static int Main(string[] args)
+        {
+            // This test validates that the rundown events are present
```

Comment:
If I follow correctly it tests if any rundown event is present, but there is no guarantee from this test that all expected rundown events are present. There have been a handful of bugs lately that involved some but not all rundown events missing. This test doesn't necessarily need to be the thing which handles that case, but it is good to have an overall test plan (typically this is just a written down prioritized list of test cases) so that everyone can agree on:
a) what tests scenarios do we intend to cover
b) which test(s) is responsible for covering each test scenario

Reply:
The rundown validation test is intended to validate the contents of the rundown while this test is intended to simply check that the rundown is supplied as part of a trace (along with other specified providers).  The list that I have been working off of is (semi) documented in #25090.